src/deploy/osp_deployer/settings/config.py

```diff
@@ -1,528 +1,561 @@
+            '/mgmt/deploy-dashboard-vm.py'
+
+        self.undercloud_conf = self.foreman_configuration_scripts +\
+            '/pilot/undercloud.conf'
+        self.install_director_sh = self.foreman_configuration_scripts +\
+            '/pilot/install-director.sh'
+        self.deploy_overcloud_sh = self.foreman_configuration_scripts + \
+            '/pilot/deploy-overcloud.py'
+        self.assign_role_py = self.foreman_configuration_scripts +\
+            '/pilot/assign_role.py'
+        self.network_env_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/network-environment.yaml'
+        self.dell_storage_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/dell-cinder-backends.yaml'
+        self.dell_env_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/dell-environment.yaml'
+        self.neutron_ovs_dpdk_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/neutron-ovs-dpdk.yaml'
+        self.static_ips_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/static-ip-environment.yaml'
+        self.static_vip_yaml = self.foreman_configuration_scripts + \
+            '/pilot/templates/static-vip-environment.yaml'
+        self.sanity_ini = self.foreman_configuration_scripts + \
+            '/pilot/deployment-validation/sanity.ini'
+        self.ipxe_rpm = self.foreman_configuration_scripts + \
+            '/pilot/ipxe/ipxe-bootimgs-20151005-1.git6847232.el7.' \
+            'test.noarch.rpm'
+
+        # The NIC configurations settings are validated after the Settings
+        # class has been instanciated.  Guard against the case where the two
+        # fixed are missing here to prevent an exception before validation
+        nics_settings = self.get_nics_settings()
+        if 'nic_env_file' in nics_settings:
+            self.nic_env_file = nics_settings['nic_env_file']
+            self.nic_env_file_path = self.foreman_configuration_scripts + \
+                '/pilot/templates/nic-configs/' + self.nic_env_file
+        if 'sah_bond_opts' in nics_settings:
+            self.sah_bond_opts = nics_settings['sah_bond_opts']
+
+        # This particular section has been moved right after the nics_settings
+        # section in order to catch the mode used by the nic-environment file
+        dellnfv_settings = self.get_settings_section(
+            "Dell NFV Settings")
+        self.ovs_dpdk_enable = dellnfv_settings[
+            'ovs_dpdk_enable']
+        self.ovs_dpdk_role_list = dellnfv_settings[
+            'ovs_dpdk_role_list']
+        self.ovs_dpdk_policy = dellnfv_settings[
+            'ovs_dpdk_policy']
+        self.enable_ovs_dpdk = False
+        if self.ovs_dpdk_enable.lower() == 'false':
+            pass
+        elif self.ovs_dpdk_enable.lower() == 'true':
+            self.enable_ovs_dpdk = True
+            for each in re.split(r'[_/]', self.nic_env_file):
```

Comment:
Hrm...  I don't think the above line is working.  It will try to split the value of nic_env_file up by occurrences of "_/", but nic_env_file looks like: "dpdk_mode1_4_port/nic_environment.yaml", which doesn't have any occurrences of that, so each will be "dpdk_mode1_4_port/nic_environment.yaml".  The line below will succeed in finding "mode", but each[-1:] will return the "l" from "yaml".
I think you want something like:
match=re.search('mode(\d)', self.nic_env_file)
if match is None:
    raise some_exception_here_due_to_no_match
self.ovs_dpdk_mode = match.group(1)

Reply:
the line does work actually.
see below:
>>> re.split(r'[/_]', 'dpdk_mode1_4_port/nic_environment.yaml')
['dpdk', 'mode1', '4', 'port', 'nic', 'environment.yaml']

Reply:
Ah, I see.  It's splitting on / OR _.

Reply:
Bingo!